lib/pinata/PinataCache.ts

```diff
@@ -72,7 +72,11 @@ class PinataCache {
          * upload the IPFS blob ourself, in which case, we'll have to request it from
          * the Gateway
          */
-        await this.cache?.add(new Request(HASH_URL));
+        if (!this.cache) {
```

Comment:
this prevents an infinite loop as `getCacheObject` is called from `setCacheObject` (which is called from `getCacheObject`)